src/main/java/com/ably/kafka/connect/ChannelConfig.java

```diff
@@ -0,0 +1,76 @@
+package com.ably.kafka.connect;
+
+import io.ably.lib.types.AblyException;
+import io.ably.lib.types.ChannelMode;
+import io.ably.lib.types.ChannelOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.ably.kafka.connect.ChannelSinkConnectorConfig.CHANNEL_CONFIG;
+import static com.ably.kafka.connect.ChannelSinkConnectorConfig.CLIENT_CHANNEL_CIPHER_KEY;
+import static com.ably.kafka.connect.ChannelSinkConnectorConfig.CLIENT_CHANNEL_PARAMS;
+
+interface ChannelConfig {
+    String getName();
+
+    ChannelOptions getOptions() throws ChannelSinkConnectorConfig.ConfigException;
+}
+
+class ChannelConfigImpl implements ChannelConfig {
+    private final ChannelSinkConnectorConfig sinkConnectorConfig;
+
+    public ChannelConfigImpl(ChannelSinkConnectorConfig sinkConnectorConfig) {
+        this.sinkConnectorConfig = sinkConnectorConfig;
+    }
+
+    @Override
+    public String getName() {
+        return sinkConnectorConfig.getString(CHANNEL_CONFIG);
+    }
+
+    public ChannelOptions getOptions() throws ChannelSinkConnectorConfig.ConfigException {
+        final Logger logger = LoggerFactory.getLogger(ChannelSinkConnectorConfig.class);
+        ChannelOptions opts;
+        String cipherKey = sinkConnectorConfig.getString(CLIENT_CHANNEL_CIPHER_KEY);
+
+        if (cipherKey != null && !cipherKey.trim().isEmpty()) {
```

Comment:
I'm not sure this code should be this liberal. If the developer configuring has specified `null` then we know that they explicitly don't want to supply a `cipherKey`. An empty string, or a string with just 'spaces', is more likely to be an unintentional mistake upstream so I would suggest it would be preferable to allow that to propagate down and fail elsewhere. i.e. reduce this check to just:

```suggestion
        if (cipherKey != null) {
```

Reply:
You are right, in Kafka connect configurations, you either provide a configuration or you do not. I have made change https://github.com/ably/kafka-connect-ably/pull/50/commits/7ea73ec0b1afadd78070ee9be4f440a1808b71f1